queue.c

```diff
@@ -306,8 +311,12 @@ BaseType_t xQueueGenericReset( QueueHandle_t xQueue,
             vListInitialise( &( pxQueue->xTasksWaitingToSend ) );
             vListInitialise( &( pxQueue->xTasksWaitingToReceive ) );
         }
+        taskEXIT_CRITICAL();
+    }
+    else
+    {
+        configASSERT( pdFAIL );

```

Comment:
The problem with asserting on "pdFAIL" is that the condition is always true so this will generate compiler warnings.  That is - the assert test will always trigger (which is the intention of course), and there is no ASSERT_ALWAYS() macro.

The previous version of this function only asserted when pxQueue was NULL, and that assert still exists at the top of the file, so that behaviour has not changed.  Now however the code will also assert if either of the two additional checks fail.  Perhaps introduce a status variable that is set to pdPASS in the happy path, and pdFAIL on what is currently line 318 - then assert on and return the status variable at the end of the function:

```
else
{
    xReturn = pdFAIL;
}

configASSERT( xReturn != pdFAIL );
return xReturn;

```

Reply:
changed